sympy/functions/elementary/miscellaneous.py

```diff
@@ -308,10 +308,10 @@ def real_root(arg, n=None):
     from sympy.functions.elementary.piecewise import Piecewise
     if n is not None:
         return Piecewise(
-            (root(arg, n), Or(Eq(n, S.One), Eq(n, S.NegativeOne))),
-            (sign(arg)*root(Abs(arg), n), And(Eq(im(arg), S.Zero),
+            (root(arg, n, evaluate=evaluate), Or(Eq(n, S.One), Eq(n, S.NegativeOne))),
+            (sign(arg)*root(Abs(arg), n, evaluate=evaluate), And(Eq(im(arg), S.Zero),
```

Comment:
Same here, multiplication by sign(arg) evaluates the expression. Use `Mul(sign(arg), root(Abs(arg), n, evaluate=evaluate), evaluate=evaluate)`  to control this.